src/operators.jl

```diff
@@ -858,3 +855,26 @@ function merge_vertices!(g::Graph{T}, vs::Vector{U} where U <: Integer) where T
 
     return new_vertex_ids
 end
+
+# special case for digraphs
+function merge_vertices!(g::AbstractGraph, vs::Vector{U} where U <: Integer)
+    @assert is_directed(g)
+    vs = sort(vs, rev=true)
+    v0 = vs[end]
+    new_vertex_ids = sort(collect(vertices(g)))
```

Comment:
The vertices will already be sorted in all current graph types. Can you make sure that a `sort(collect(range))` operation is compiled into a NOP?